pkg/httpclient/http_client.go

```diff
@@ -7,24 +7,29 @@ import (
 	"time"
 )
 
-type HTTPClient struct {
+type Interface interface {
```

Comment:
Nitpick: the common convention in Go is to [name interfaces to describe what they do](https://go.dev/doc/effective_go#interface-names). I'd name it something similar to `httpclient.Getter`.

Reply:
It's "coincidentally" just a getter, it's actually an interface for HTTP client (could be POST / PUT as well, eventually). I wanted to name it "HTTPClientInterface" at first, but then linter complained that I'm repeating "httpclient.HTTPClientInterface" .. and I thought "httpclient.Interface" was not so bad after all, if we read it entirely with the package prefix...

Reply:
I'd prefer to rename to `HTTPClient` because it's more aligned with what this interface is about, however it would make us write `httpclient.HTTPClient` which isn't pretty either